webClient/src/app/editor/code-editor/monaco/monaco.service.ts

```diff
@@ -304,15 +304,15 @@ export class MonacoService implements OnDestroy {
     }
 
     const _editor = this.editorControl.editorCore.getValue().editor;
-    const previousModel = _editor.getModel(this.generateUri(this.previousFileContents.model));
+    const previousModel = monaco.editor.createModel(this.previousFileContents.model.contents);
```

Comment:
By creating a new model and not using an existing one, the comparison views in the Diff viewer during normal operation no longer synchronizes with the Editor. This means that when we edit two files in the normal Editor, we don't track its state: 

the "changed" indicator no longer works

![image](https://user-images.githubusercontent.com/20528015/141224315-ad1c846f-b335-436d-8bb8-8cddf17254d3.png)


pressing Ctrl + S no longer works as intended (it continues to save a file, but it saves your last opened file, it doesn't properly save file A to file B when you switch from left-hand side and right-hand side)

If you can't get it to work cleanly, my recommendation is to duplicate and change the spawnDiffViewer method into a new method (spawnDatasetComparisonViewer ?)